src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -178,32 +178,39 @@ public Task IndexAsync(Package package, CancellationToken cancellationToken)
             var frameworks = GetCompatibleFrameworksOrNull(framework);
             var search = (IQueryable<Package>)_context.Packages.Where(p => p.Listed);
 
-            if (!string.IsNullOrEmpty(query))
+            IQueryable<Package> ComposePackageQuery(IQueryable<Package> packageQuery)
             {
-                query = query.ToLower();
-                search = search.Where(p => p.Id.ToLower().Contains(query));
-            }
+                if (!string.IsNullOrEmpty(query))
+                {
+                    query = query.ToLower();
+                    packageQuery = packageQuery.Where(p => p.Id.ToLower().Contains(query));
```

Comment:
We only need to filter on `query` once, could you move lines 183-187 out of `ComposePackageQuery` (or `AddSearchFilters`)? I would make line 179 something like:

```csharp
var search = AddSearchFilters(_context.Packages);
if (!string.IsNullOrEmpty(query))
{
  query = query.ToLower();
  search = search.Where(p => p.Id.ToLower().Contains(query));
}

var packageIds = ...
```